### PR TITLE
Issue 2310: Adds the autorecovery status service.

### PR DESCRIPTION
**Pull request by fantapsody**
Descriptions of the changes in this PR:

### Motivation

It would be convenient to support enable, disable, or check the status of autorecovery with REST API, just like what does in the toggle command.

### Changes

Adds an API to enable, disable, and get the current status of autorecovery at `/api/v1/autorecovery/status`.

Master Issue: #2310 

> ---
> In order to uphold a high standard for quality for code contributions, Apache BookKeeper runs various precommit
> checks for pull requests. A pull request can only be merged when it passes precommit checks.
>
> ---
> Be sure to do all of the following to help us incorporate your contribution
> quickly and easily:
>
> If this PR is a BookKeeper Proposal (BP):
>
> - [ ] Make sure the PR title is formatted like:
>     `<BP-#>: Description of bookkeeper proposal`
>     `e.g. BP-1: 64 bits ledger is support`
> - [ ] Attach the master issue link in the description of this PR.
> - [ ] Attach the google doc link if the BP is written in Google Doc.
>
> Otherwise:
> 
> - [x] Make sure the PR title is formatted like:
>     `<Issue #>: Description of pull request`
>     `e.g. Issue 123: Description ...`
> - [ ] Make sure tests pass via `mvn clean apache-rat:check install spotbugs:check`.
> - [x] Replace `<Issue #>` in the title with the actual Issue number.
> 
> ---

### CODE REVIEW

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/server/http/BKHttpServiceProvider.java [hunk 1]
```diff
@@ -199,7 +200,7 @@ public HttpEndpointService provideHttpEndpointService(ApiType type) {
                 return new DeleteLedgerService(configuration);
             case LIST_LEDGER:
                 return new ListLedgerService(configuration, bookieServer);
-            case GET_LEDGER_META:
+            case  GET_LEDGER_META:
```
Comment (eolivelli) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/server/http/BKHttpServiceProvider.java:203]: Nit: remove extra space

**Comment by fantapsody** [comment]
I have added unit tests, please take another look, thanks! @eolivelli

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/server/http/service/AutoRecoveryStatusService.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+                ledgerUnderreplicationManager.isLedgerReplicationEnabled()));
+        return new HttpServiceResponse(body, HttpServer.StatusCode.OK);
+    }
+
+    private HttpServiceResponse handlePutStatus(HttpServiceRequest request,
+                                                LedgerUnderreplicationManager ledgerUnderreplicationManager)
+            throws Exception {
+        Map<String, String> params = ObjectUtils.defaultIfNull(request.getParams(), Collections.emptyMap());
+        String enabled = params.get("enabled");
+        if (enabled == null) {
```
Comment (mino181295) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/server/http/service/AutoRecoveryStatusService.java:88]: Extra check if equals "true" or "false"?

**Comment by mino181295** [comment]
Just a question, LGTM

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/server/http/service/AutoRecoveryStatusService.java (hunk 2 above)
Comment (eolivelli) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/server/http/service/AutoRecoveryStatusService.java:88]: @mino181295 we are evaluating the value at line 92

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/server/http/service/AutoRecoveryStatusService.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            return new HttpServiceResponse("Param 'enabled' not found in " + params,
+                    HttpServer.StatusCode.INTERNAL_ERROR);
```
Comment (eolivelli) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/server/http/service/AutoRecoveryStatusService.java:90]: What about 'Bad request'?

**Comment by eolivelli** [comment]
Just last minor comment from me.
Then we ate good to go

@jiazhai PTAL

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/server/http/service/AutoRecoveryStatusService.java (hunk 3 above)
Comment (fantapsody) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/server/http/service/AutoRecoveryStatusService.java]: That would be much better, just changed it, thanks!

**Comment by eolivelli** [comment]
can you please merge with current master?

**Comment by fantapsody** [comment]
> can you please merge with current master?

Sure, I have merged with current master, please take a look.

**Pull request merged**